### DISCUSSION TITLE
sdiag data available?

### DISCUSSION THREAD

**Discussion by robgics** [PySlurm/pyslurm#320] [General]
We're using pyslurm solely for data capture to be able to easily view trends in grafana.  It's working great for job and node data.  But now we're looking to get some of the sdiag data out of slurm to get a view of overall performance.  I cannot find an interface in pyslurm to get that.  Am I missing it, or is it planned to be added, or something else?

Thanks.

**Comment by tazend** [comment]
Hi @robgics 

Yes, this is available: https://pyslurm.github.io/23.2/reference/statistics/

As the documentation states on that page, this is for now just available in the "old" API, and will also be reworked at some point (like the Job or Node classes for example, which are already done) -  That shouldn't be too hard, and I have already some other `slurmctld` related stuff in the works, so I'll probably get to that soon.

So for now, you can do:

```python
import pyslurm
stats = pyslurm.statistics().get()
```

`stats` is then a `dict` which contains all the information from `sdiag`. Since the data in this `dict` is currently not documented in pyslurm, you would need to explore yourselves what it contains after fetching, or checking out the source code: https://github.com/PySlurm/pyslurm/blob/main/pyslurm/pyslurm.pyx#L4673

As said: it is planned to rework this and move away from returning a dict with undocumented data, and rather use properly documented instance attributes. But using `pyslurm.statistics().get()` for now should get you the data you'll need.

**Reply by robgics** [reply]
Excellent!  I had just grepp'd for sdiag through the code and didn't find anything that way, or any promising sounding python filenames.  I'm glad I was wrong, I didn't want my own homebrew'd code AND pyslurm to keep track of.
